packages/react/lib/common.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+exports.createCombinedContext = function(
+  config,
+  ReactContext,
+  combineContexts
+) {
+  var mergedConfig = Object.assign({}, config);
+  delete mergedConfig.extensions;
+  var mergedContexts = [ReactContext];
+  if (config.extensions) {
+    config.extensions.forEach(function(extension) {
+      var context = extension.context;
+      mergedContexts.push(context);
+      var extensionConfig = extension.config;
+      mergedConfig = Object.assign(mergedConfig, extensionConfig);
```

Comment:
I think I would rather `bind` the extension specific options to their respective constructor, something like..:

```
var config = Object.assign({}, mergedConfig, extensionConfig);
mergedContexts.push(
  Function.prototype.bind.call(context, context, config)
);
```

Reply:
...but that is clearly suboptimal, as well (cloning...)

Reply:
Meanwhile, I have come to think that your initial implementation is the best we can do without changing the way `mixinable` works.